pkg/cri/server/helpers_linux.go

```diff
@@ -141,8 +140,14 @@ func checkSelinuxLevel(level string) error {
 	return nil
 }
 
+// apparmorEnabled returns true if apparmor is enabled, supported by the host,
+// and if apparmor_parser is installed, and if we're not running docker-in-docker..
 func (c *criService) apparmorEnabled() bool {
-	return runcapparmor.IsEnabled() && !c.config.DisableApparmor
+	// see https://github.com/docker/docker/commit/de191e86321f7d3136ff42ff75826b8107399497
+	if !c.config.DisableApparmor || os.Getenv("container") != "" {
```

Comment:
FWIW; I was in doubt wether to add the `os.Getenv()` here, or in `hostSupportsAppArmor()`; happy to move it there